metadata-ingestion/src/datahub/ingestion/source/gcs/gcs_source.py

```diff
@@ -41,6 +59,13 @@ class GCSSourceConfig(DatasetSourceConfigMixin, PathSpecsConfigMixin):
         description="Maximum number of rows to use when inferring schemas for TSV and CSV files.",
     )
 
+    number_of_files_to_sample: int = Field(
+        default=100,
+        description="Number of files to list to sample for schema inference. This will be ignored if sample_files is set to False in the pathspec.",
+    )
```

Comment:
Seems more efficient to have this be an optional int, rather than have this and `sample_files` bool. Also, what do you think about the name `file_sample_size` or something like that?